qiskit_experiments/library/driven_freq_tuning/coefficient_utils.py

```diff
@@ -0,0 +1,234 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Coefficients characterizing Stark shift."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import numpy as np
+
+from qiskit.providers.backend import Backend
+from qiskit_ibm_experiment.service import IBMExperimentService
+from qiskit_ibm_experiment.exceptions import IBMApiError
+
+from qiskit_experiments.framework.backend_data import BackendData
+from qiskit_experiments.framework.experiment_data import ExperimentData
+
+
+@dataclass
+class StarkCoefficients:
+    """A dataclass representing a set of coefficients characterizing Stark shift."""
+
+    pos_coef_o1: float
```

Comment:
For now I think either 3rd or 4th order is fine. Usually higher order terms result in the overfitting. Since coefficients couple to the rest of util functions, support of function at any order needs complicated logic and potentially introduce edge case.

Reply:
Perhaps I misread the context. If you are speaking about the data structure, I think storing two lists for positive and negative coefficients are also okey.

Reply:
If you feel good at keeping the order at 3, then there is not much advantage to changing to a list versus keeping each coeficient as a separate attribute. I was just checking if you thought there could be a need for flexibility. Then the list could be better.